runtime/sema/check_composite_declaration.go

```diff
@@ -2074,9 +2081,21 @@ func (checker *Checker) checkDefaultDestroyEventParam(
 
 	// make `self` and `base` available when checking default arguments so the fields of the composite are available
 	// as this event is emitted when the resource is destroyed, these values should be fully entitled
-	fullyEntitledAccess := NewAccessFromEntitlementSet(containerType.SupportedEntitlements(), Conjunction)
-	checker.declareSelfValue(fullyEntitledAccess, containerType, containerDeclaration.DeclarationDocString())
-	if compositeContainer, isComposite := containerType.(*CompositeType); isComposite && compositeContainer.Kind == common.CompositeKindAttachment {
+	supportedEntitlements := containerType.SupportedEntitlements()
+	fullyEntitledAccess := UnauthorizedAccess
+	if supportedEntitlements != nil && supportedEntitlements.Len() > 0 {
+		fullyEntitledAccess = NewAccessFromEntitlementSet(supportedEntitlements, Conjunction)
+	}
```

Comment:
Seems this pattern is/has-to-be used every time the constructor `NewAccessFromEntitlementSet` is called. Would it makes sense to move this logic to the `NewAccessFromEntitlementSet` itself?

Reply:
Mostly just tried to make the code handle all cases and make it consistent, leaving existing cases as-is, but yeah, we can also simplify the case where `NewAccessFromEntitlementSet` is called. It already handles `Len() == 0`, but we can also add the nil check there.

Reply:
@SupunS Pushed a few commits to improve this